bundles/org.openhab.binding.souliss/src/main/java/org/openhab/binding/souliss/handler/SoulissT19Handler.java

```diff
@@ -0,0 +1,179 @@
+                        commandSEND(SoulissBindingProtocolConstants.SOULISS_T1N_SET,
+                                (byte) (((PercentType) command).shortValue() * 255.00 / 100.00));
+                    } else if (command.equals(OnOffType.ON)) {
+                        commandSEND(SoulissBindingProtocolConstants.SOULISS_T1N_ON_CMD);
+
+                    } else if (command.equals(OnOffType.OFF)) {
+                        commandSEND(SoulissBindingProtocolConstants.SOULISS_T1N_OFF_CMD);
+                    }
+                    break;
+
+                case SoulissBindingConstants.ROLLER_BRIGHTNESS_CHANNEL:
+                    if (command.equals(UpDownType.UP)) {
+                        commandSEND(SoulissBindingProtocolConstants.SOULISS_T1N_BRIGHT_UP);
+                    } else if (command.equals(UpDownType.DOWN)) {
+                        commandSEND(SoulissBindingProtocolConstants.SOULISS_T1N_BRIGHT_DOWN);
+                    }
+                    break;
+                case SoulissBindingConstants.SLEEP_CHANNEL:
+                    if (command instanceof OnOffType) {
+                        commandSEND((byte) (SoulissBindingProtocolConstants.SOULISS_T1N_TIMED + xSleepTime));
+                    }
+                    break;
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.ONLINE);
+        gwConfigurationMap = thingGeneric.getConfiguration();
+        if (gwConfigurationMap.get(SoulissBindingConstants.SLEEP_CHANNEL) != null) {
+            xSleepTime = ((BigDecimal) gwConfigurationMap.get(SoulissBindingConstants.SLEEP_CHANNEL)).byteValue();
+        }
+        if (gwConfigurationMap.get(SoulissBindingConstants.CONFIG_SECURE_SEND) != null) {
+            bSecureSend = ((Boolean) gwConfigurationMap.get(SoulissBindingConstants.CONFIG_SECURE_SEND)).booleanValue();
+        }
+    }
+
+    public void setState(@Nullable PrimitiveType state) {
+        super.setLastStatusStored();
+        if (state != null) {
+            updateState(SoulissBindingConstants.SLEEP_CHANNEL, OnOffType.OFF);
+            logger.debug("T19, setting state to {}", state.toFullString());
+            this.updateState(SoulissBindingConstants.ONOFF_CHANNEL, (OnOffType) state);
+        }
+    }
+
+    public void setRawStateDimmerValue(byte dimmerValue) {
+        try {
+            if (dimmerValue != t1nRawStateByte0) {
+                logger.debug("T19, setting dimmer to {}", dimmerValue);
+                updateState(SoulissBindingConstants.DIMMER_BRIGHTNESS_CHANNEL,
+                        PercentType.valueOf(String.valueOf(Math.round((t1nRawStateByte0 / 255) * 100))));
+            }
+        } catch (IllegalStateException ex) {
+            logger.debug("UUID: {}", this.getThing().getUID().getAsString());
+            logger.debug(" - Update state error (in setDimmerValue): {}", ex.getMessage());
```

Comment:
```suggestion
            logger.debug("UUID: {}, had an update dimmer state error:{}", this.getThing().getUID().getAsString(), ex.getMessage());
```

Save CPU by only doing a check on if the debug mode is enabled once. Alternatively you can check in an if statement is debug is enabled.

Reply:
fixed